core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -428,6 +471,39 @@ class TiBatchWrite(@transient val df: DataFrame,
     }
   }
 
+  private def generateDataToBeRemovedRdd(rdd: RDD[WrappedRow], startTs: TiTimestamp) = {
+    rdd
+      .map { wrappedRow =>
```

Comment:
change `rdd.map` to `rdd.mapPartitions`, cause if use `rdd.map` the following code will be called `rdd.size` times

`val snapshot = TiSessionCache.getSession(tiConf).createSnapshot(startTs)`

Reply:
It is planned to be done when we support batch get.

Reply:
it can be done without batch get.